wasi_snapshot_preview1/clock.go

```diff
@@ -0,0 +1,123 @@
+package wasi_snapshot_preview1
+
+import (
+	"context"
+	"time"
+
+	"github.com/tetratelabs/wazero/api"
+)
+
+const (
+	// functionClockResGet returns the resolution of a clock.
+	// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-clock_res_getid-clockid---errno-timestamp
+	functionClockResGet = "clock_res_get"
+
+	// importClockResGet is the WebAssembly 1.0 Text format import of functionClockResGet.
+	importClockResGet = `(import "wasi_snapshot_preview1" "clock_res_get"
+    (func $wasi.clock_res_get (param $id i32) (param $result.resolution i32) (result (;errno;) i32)))`
+
+	// functionClockTimeGet returns the time value of a clock.
+	// See https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-clock_time_getid-clockid-precision-timestamp---errno-timestamp
+	functionClockTimeGet = "clock_time_get"
+
+	// importClockTimeGet is the WebAssembly 1.0 Text format import of functionClockTimeGet.
+	importClockTimeGet = `(import "wasi_snapshot_preview1" "clock_time_get"
+    (func $wasi.clock_time_get (param $id i32) (param $precision i64) (param $result.timestamp i32) (result (;errno;) i32)))`
+)
+
+// https://github.com/WebAssembly/WASI/blob/snapshot-01/phases/snapshot/docs.md#-clockid-enumu32
+const (
+	// clockIDRealtime is the clock ID named "realtime" associated with sys.Walltime
+	clockIDRealtime = iota
+	// clockIDMonotonic is the clock ID named "monotonic" with sys.Nanotime
+	clockIDMonotonic
+	// clockIDProcessCputime is the unsupported clock ID named "process_cputime_id"
```

Comment:
these values are new. mainly, changes in this file are validation only and fix to return ErrnoNotsup on unsupported